tests/functional/base_test.go

```diff
@@ -94,37 +96,66 @@ func CreateOVNDBCluster(namespace string, OVNDBClusterName string, spec map[stri
 	return th.CreateUnstructured(raw)
 }
 
+func UpdateOVNDBCluster(cluster *ovnv1.OVNDBCluster) {
+	k8sClient.Update(ctx, cluster)
+}
+
 func OVNDBClusterConditionGetter(name types.NamespacedName) condition.Conditions {
 	instance := GetOVNDBCluster(name)
 	return instance.Status.Conditions
 }
 
+func ScaleDBCluster(name types.NamespacedName, replicas int32) {
+	Eventually(func(g Gomega) {
+		c := GetOVNDBCluster(name)
+		*c.Spec.Replicas = replicas
+		g.Expect(k8sClient.Update(ctx, c)).Should(Succeed())
+	}).Should(Succeed())
+}
+
 // CreateOVNDBClusters Creates NB and SB OVNDBClusters
-func CreateOVNDBClusters(namespace string, nad string) []types.NamespacedName {
+func CreateOVNDBClusters(namespace string, nad map[string][]string, replicas int32) []types.NamespacedName {
 	dbs := []types.NamespacedName{}
 	for _, db := range []string{v1beta1.NBDBType, v1beta1.SBDBType} {
 		name := fmt.Sprintf("ovn-%s", uuid.New().String())
 		spec := GetDefaultOVNDBClusterSpec()
+		stringNad := ""
+		// OVNDBCluster doesn't allow multiple NADs, hence map len
+		// must be <= 1
+		Expect(len(nad)).Should(BeNumerically("<=", 1))
+		for k, _ := range nad {
+			if strings.Contains(k, "/") {
+				// k = namespace/nad_name, split[1] will return nad_name (e.g. internalapi)
+				stringNad += strings.Split(k, "/")[1]
+			}
+		}
 		spec["dbType"] = db
-		spec["networkAttachment"] = nad
+		spec["networkAttachment"] = stringNad
+		spec["replicas"] = replicas
 
 		instance := CreateOVNDBCluster(namespace, name, spec)
 
 		instance_name := types.NamespacedName{Name: instance.GetName(), Namespace: instance.GetNamespace()}
 
-		dbaddr := "tcp:10.1.1.1:6641"
+		db_name := "nb"
 		if db == v1beta1.SBDBType {
-			dbaddr = "tcp:10.1.1.1:6642"
+			db_name = "sb"
 		}
-		// the Status field needs to be written via a separate client
+		statefulSetName := types.NamespacedName{
+			Namespace: instance.GetNamespace(),
+			Name:      "ovsdbserver-" + db_name,
+		}
+		th.SimulateStatefulSetReplicaReadyWithPods(
+			statefulSetName,
+			nad,
+		)
+		// Ensure that PODs are ready and DBCluster have been reconciled
+		// with all information (Status.DBAddress and internalDBAddress
+		// are set at the end of the reconcileService)
 		Eventually(func(g Gomega) {
 			ovndbcluster := GetOVNDBCluster(instance_name)
-			ovndbcluster.Status.InternalDBAddress = dbaddr
-			if nad != "" {
-				ovndbcluster.Status.DBAddress = dbaddr
-			}
-			g.Expect(k8sClient.Status().Update(ctx, ovndbcluster)).Should(Succeed())
-		}, timeout, interval).Should(Succeed())
+			g.Expect(ovndbcluster.GetExternalEndpoint()).ToNot(BeEmpty())
```

Comment:
I'm confused. So even when nad is empty, we still validate here that dbAddress is not empty. But I thought in this case it's always empty, no? What am I missing? Is it because GetExternalEndpoint() returns two values (string, error)?

Reply:
Oh, I think you're correct. And I think two things were wrong with this chunk.
1- As you said, since it returned (string, error) and error was not null, it was not failing with .BeEmpty
2- The whole eventually needs to be checked with .Succeed() in order to evaluate all Expect inside eventually. [Documentation](https://pkg.go.dev/github.com/onsi/gomega#Eventually)
```
Eventually also supports a special case polling function that takes a single Gomega
argument and returns no values. Eventually assumes such a function is making
assertions and is designed to work with the Succeed matcher to validate that all
assertions have passed. For example:

Eventually(func(g Gomega) {
	model, err := client.Find(1138)
	g.Expect(err).NotTo(HaveOccurred())
	g.Expect(model.Reticulate()).To(Succeed())
	g.Expect(model.IsReticulated()).To(BeTrue())
	g.Expect(model.Save()).To(Succeed())
}).Should(Succeed())
```

Fixed, done